numademo.c

```diff
@@ -301,13 +301,14 @@ int max_node, numnodes;
 void get_node_list()
 {
         int a, got_nodes = 0;
-        long free_node_sizes;
+        long free_node_sizes, free_mem;
 
         numnodes = numa_num_configured_nodes();
         node_to_use = (int *)malloc(numnodes * sizeof(int));
         max_node = numa_max_node();
         for (a = 0; a <= max_node; a++) {
-                if(numa_node_size(a, &free_node_sizes) != -1)
```

Comment:
How about this instead?

```c
if (numa_node_size(a, &free_node_sizes) > 0)
```

(Nitpick: we usually do a space between `if` (or `for`, `while`, etc.) and the initial `(`, also to differentiate them from function calls.)